client/packages/programs/src/JsonForms/common/components/Text.tsx

```diff
@@ -71,7 +73,7 @@ const usePatternValidation = (
   const { customError, setCustomError } = useJSONFormsCustomError(path, 'Text');
 
   useEffect(() => {
-    if (!pattern || !value) {
+    if (!pattern || value === undefined) {
```

Comment:
This was a bug that I noticed along the way... i.e. an empty string `!string` is evaluating to true.

Note, we shouldn't really do these `!!` or `!` checks but do a proper check against `undefined` instead. This is much clearer, and you don't have to worry about truthy and falsy conversions...

Reply:
re the `!!` or `!` sometimes this is valid. I have had cases where I deliberately want to check `null` `undefined` or `` and prefer to have the terse code. You are right though, it does require the coder to know about the js truthy and falsy conversions which have tripped people up. Is it my responsibility that other coders don't know things though? should I write longhand code just in case?

Reply:
Personally I use it to also make clear what my indention was, e.g. to avoid wondering if I didn't thought about the truthy/falsy case or if it was on purpose. I would even do this `if (value !== undefined && value !== '')` for `value: string | undefined`, i.e. first narrow the type down to a `string` and then compare the string.